js/data.js

```diff
@@ -6,3 +6,18 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var form = document.querySelector('.form');
+
+form.addEventListener('reset', beforeReset);
+function beforeReset(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('data', dataJSON);
+}
+
+var previousDataJSON = localStorage.getItem('data');
+if (previousDataJSON !== null) {
+  data.entries.push(JSON.parse(previousDataJSON));
```

Comment:
You are adding the entire previous `data` object as a value in the `entries` property of the `data` object. This is not what you intended to do, I'm sure.